runtime/j9vm/jvm.c

```diff
@@ -6465,3 +6515,35 @@ JVM_BeforeHalt()
 {
 	/* To be implemented via https://github.com/eclipse-openj9/openj9/issues/1459 */
 }
+
+#if defined(J9VM_ZOS_3164_INTEROPERABILITY) && (JAVA_SPEC_VERSION >= 17)
+/*
+ * Utility function used by NativeLibraries to invoke JNI_OnLoad or JNI_OnUnload
+ * functions in 31-bit native interoperability targets. This requires mapping to
+ * the corresponding 31-bit JavaVM object handle, along with invoking CEL4RO31
+ * (via FFI) to the corresponding target function.
+ *
+ * @param vm The JavaVM pointer first parameter for JNI_OnXLoad function
+ * @param handle The target function pointer to invoke - should be a 31-bit interop target
+ * @param isOnLoad TRUE if invoking JNI_OnLoad, FALSE if invoking JNI_OnUnload
+ * @param reserved The reserved second parameter for JNI_OnXLoad function
+ *
+ * @return the return value for JNI_OnLoad, or 0 for JNI_OnUnload
+ */
+jint JNICALL
+JVM_Invoke31BitJNI_OnXLoad(JavaVM *vm, void *handle, jboolean isOnLoad, void *reserved)
+{
+	J9JavaVM *javaVM = (J9JavaVM *)vm;
+	jint result = isOnLoad ? JNI_VERSION_1_1 : 0;
+
+	Trc_SC_Invoke31BitJNI_OnXLoad_Entry(vm, handle, isOnLoad, reserved);
+
+	if (J9_IS_31BIT_INTEROP_TARGET(handle)) {
+		result = BFUjavaVM->internalVMFunctions->invoke31BitJNI_OnXLoad(javaVM, handle, isOnLoad, reserved);
```

Comment:
This should use `javaVM` instead of `BFUjavaVM`.

Reply:
Fixed in 55ae9d3